owncloudApp/src/main/java/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -508,14 +597,19 @@ private void initToolbar(String current_dir) {
 
         ActionBar actionBar = getSupportActionBar();
 
-        boolean notRoot = (mParents.size() > 1);
-
         if (actionBar != null) {
-            actionBar.setDisplayHomeAsUpEnabled(notRoot);
-            actionBar.setHomeButtonEnabled(notRoot);
+            actionBar.setDisplayHomeAsUpEnabled(true);
+            actionBar.setHomeButtonEnabled(true);
```

Comment:
Why did this became `true` always? 😮

Reply:
Because when we enter a folder we will always want it to go backwards. Here are new cases:

- In the spaces view the back is not operational.
- In the file overview there are two options:
      1.     It has spaces, so we can go backwards
      2.     It has no spaces, we can't go back.
- It has also been taken into account that in oc10, the last route where a share was made is saved. Therefore more logic has been added at this point.

Most of this logic is controlled with the fucntion `updateToolbar` in the different flows of the class.